### ISSUE TITLE
Oauth2 authentication

### ISSUE THREAD

**Issue by sramsay335i** [geral2/SQL-APIConsumer#41]
Hi Geral, 
  Is it possible to do Oauth 2.0 Authorization with API Caller?   I tried the sample code that calls APICaller_GETAuth but not getting anything.
Thanks,
Sean

**Comment by geral2** [comment]
Hi @sramsay335i,

I think it's possible, Could you try using the procedure named  **dbo.APICaller_POST_Encoded**;

```
DECLARE @wurl                     VARCHAR(MAX)	= 'https://xxx/oauth/token' 
	   ,@header			NVARCHAR(MAX)	= '[]' 
	   ,@grant_type		NVARCHAR(1000)	= 'client_credentials' 
	   ,@client_id		        NVARCHAR(1000)	= 'xxxx' 
	   ,@client_secret             NVARCHAR(MAX)	= 'xxxx' 
	   ,@audience		        NVARCHAR(MAX)	= 'https://api.xx.com' 

DECLARE @Body nvarchar(max)	= CONCAT('grant_type=',@grant_type,'&client_id=',@client_id,'&client_secret=',@client_secret,'&audience=',@audience)

DECLARE @Results AS TABLE
(
	Context varchar(max)
)

DECLARE @Result AS VARCHAR(MAX)

--INSERT INTO @Results

EXEC dbo.APICaller_POST_Encoded  @URL		= @wurl
								,@Headers	= @header
								,@JsonBody	= @Body


```

Let me know if it works?

**Comment by sramsay335i** [comment]
Hi Geral,
  Thanks for this.  Had no idea.   I think I need to upgrade my API_Consumer assembly because when I run the CREATE PROC for APICaller_POST_Encoded , it says it can't find the method.
   Do you have any upgrade documentation available?   The  following ALTER ASSEMBLY statement says it can't open the physical file: 
ALTER ASSEMBLY API_Consumer FROM 'C:\Temp\SQL-APIConsumer-2.2\API_Consumer\bin\Debug\API_Consumer.dll'
  Thanks for getting back to me, we use your API_Caller all the time 👍
Chat soon,
Sean

**Comment by geral2** [comment]
I don't really have any good documentation. But could you try removing all the SQL dependencies related to API_Consumer, then try to recreate it using API_Consumer_V2.2 Script attached in [V2.2]( https://github.com/geral2/SQL-APIConsumer/releases/tag/v2.2).

Let me know how it goes.

**Comment by sramsay335i** [comment]
Hi Geral,
  I uninstalled all the dependencies and assemblies.    I installed the latest from v2.2 and now I get a '....Assembly in host store has a different signature than assembly in GAC.' error for the Newtonsoft.Json package.    I installed it on another machien (clean install) and get the same error.   Have you seen this before?
Sean

Could not load file or assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. Assembly in host store has a different signature than assembly in GAC. (Exception from HRESULT: 0x80131050)

**Comment by geral2** [comment]
Yes, you need to make sure to install the DLLs dependecies, before creating the APIConsumer, remember to replace the database name.

It could be found in [API.Consumer.Binary.dependicies](https://github.com/geral2/SQL-APIConsumer/releases/download/v2.2/API.Consumer.Binary.dependencies.sql)

**Comment by sramsay335i** [comment]
Hi Geral,
  Finally got it working here.  I followed your instructions of the installation to no avail.    I kept getting a Newtonsoft.Json.dll version error in the GAC.   

What I had to do was delete the Newtonsoft.Json.dll from the Global Assembly Cache (GAC) using the Windows gacutil.exe utility.  
- (cmd:  gacutil /u Newtonsoft.Json.dll)   - this uninstalls ALL versions of the dll in the Global Assembly 

Then I installed the new one from your repository (cmd: gacutil /i Newtonsoft.Json.dll) . 

Once that was done, I rebooted the machine which refreshes the GAC then all was working correctly with your SQL Stored Procedures.
 
  Your instructions were still vital though because I didn't realize the assemblies had to be installed in a correct order.

Thanks for your patience and time helping me resolve this.   
Much appreciated,
Sean
